core/src/main/java/io/smallrye/concurrency/SmallRyeConcurrencyManager.java

```diff
@@ -129,6 +125,18 @@ ThreadContextProviderPlan getProviderPlan(String[] propagated, String[] unchange
 				clearedProviders.add(provider);
 		}
 
+		// lastly, add all unmentioned contexts to cleared list
+		Set<String> unmentionedContexts = this.getAllThreadContextProviderNames();
+		unmentionedContexts.removeAll(unchangedSet);
+		unmentionedContexts.removeAll(propagatedSet);
+		unmentionedContexts.removeAll(clearedSet);
+		for (String type : unmentionedContexts) {
+			ThreadContextProvider provider = providersByType.get(type);
+			if (provider != null) {
+				clearedProviders.add(provider);
+			}
+		}
+
```

Comment:
Are you sure about this behaviour? It used to be required to treat them as "unchanged". Also the docs for "cleared" says:

```
The default set of cleared thread context types is
         * {@link ThreadContext#TRANSACTION}, which means that a transaction
         * is not active on the thread when the action or task runs, such
         * that each action or task is able to independently start and end
         * its own transactional work.
```

Which to me says that we should not clear anything implicitly except TRANSACTION.

Reply:
So I though, but was proved wrong.

> Thread context types which are not otherwise included in this set or in the {@link #unchanged} set are cleared from the thread of execution for the duration of the action or task.

See https://github.com/eclipse/microprofile-concurrency/issues/74
The test linked to it was originally failing due to this problem. We then decided to correct it to test what it explicitly states but a new test for the above described behaviour will be added anyway.

Reply:
Ah OK, well that's confusing as hell, but let me reenable you fix then.